### ISSUE TITLE
Allow the ModLoader autoload to be in a different spot than first

### ISSUE THREAD

**Issue by Qubus0** [GodotModding/godot-mod-loader#257]
Issue: some games (Dome Keeper) implement cheat detection with autoloads. Placing the ModLoader after those will make modding them a lot harder.

* allow the modloader and store to be in a different spot than first
* still make sure that the store is added before the loader
* possibly add a log message that mentions that all autoloads above the mod loader won't be moddable if it is not in the first spot

this issue goes hand in hand with #250 , but is easier to implement for now.

**Comment by KANAjetzt** [comment]
Currently, checking the autoload position is the third step in the `_init()` function of the ModLoader.

```gdscript
func _init() -> void:
	# if mods are not enabled - don't load mods
	if ModLoaderStore.REQUIRE_CMD_LINE and not _ModLoaderCLI.is_running_with_command_line_arg("--enable-mods"):
		return

	# Rotate the log files once on startup. Can't be checked in utils, since it's static
	ModLoaderLog._rotate_log_file()

	# Ensure ModLoaderStore and ModLoader are the 1st and 2nd autoloads
-->	_check_autoload_positions()
	
```

So if `ModLoaderStore` is not available, we encounter the first block here: 

```gdscript
func _init() -> void:
	# if mods are not enabled - don't load mods
-->	if ModLoaderStore.REQUIRE_CMD_LINE and not _ModLoaderCLI.is_running_with_command_line_arg("--enable-mods"):
		return

	# Rotate the log files once on startup. Can't be checked in utils, since it's static
	ModLoaderLog._rotate_log_file()

	# Ensure ModLoaderStore and ModLoader are the 1st and 2nd autoloads
	_check_autoload_positions()
	
```

And even before that we currently deprecate `UNPACKED_DIR`:

```gdscript
var UNPACKED_DIR := "res://mods-unpacked/" setget ,deprecated_direct_access_UNPACKED_DIR

func deprecated_direct_access_UNPACKED_DIR() -> String:
	ModLoaderDeprecated.deprecated_message("The const \"UNPACKED_DIR\" was removed, use \"ModLoaderMod.get_unpacked_dir()\" instead", "6.0.0")
	return _ModLoaderPath.get_unpacked_mods_dir_path()
```

And `_deprecated_log(msg: String)` checks `if ModLoaderStore.ml_options.ignore_deprecated_errors`


```gdscript
# Internal func for logging with support to trigger warnings instead of fatal
# errors
static func _deprecated_log(msg: String) -> void:
	if ModLoaderStore.ml_options.ignore_deprecated_errors:
		ModLoaderLog.warning(msg, LOG_NAME)
	else:
		ModLoaderLog.fatal(msg, LOG_NAME)
```

**Comment by KANAjetzt** [comment]
closed by #264
